internal/commands/root.go

```diff
@@ -37,12 +36,19 @@ func NewRootCmd(use string, dockerCli command.Cli) *cobra.Command {
 				return cmd.GenBashCompletion(dockerCli.Out())
 			case "zsh":
 				return cmd.GenZshCompletion(dockerCli.Out())
-			case "":
-				// Actually unset
-				return nil
 			default:
-				return fmt.Errorf("%q is not a supported shell", completion)
+				if len(completion) > 0 {
+					return fmt.Errorf("%q is not a supported shell", completion)
+				}
+			}
+
+			if len(args) != 0 {
+				fmt.Printf("%q is not a docker app command", args[0])
```

Comment:
I wonder if returning this error is not enough? 🤔 We are in a `RunE` function, with a `E` postfix for error. If a command returns an error `cobra` will print it and return an error code. So maybe:
```go
return fmt.Errorf("%q is not a valid docker app command", args[0])
```
WDYT?

Reply:
We also want to print the help, not just return an error.

Reply:
IIRC there's a cobra option to print the help command?

Reply:
And we want to have the error message before the usage one to be consistent with flag error message